src/webgpu/api/operation/sampling/derivative.spec.ts

```diff
@@ -0,0 +1,216 @@
+export const description = `
+Tests the behavior of derivatives.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { assert } from '../../../../common/util/util.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+const kRTSize = 2;
+const kBytesPerRow = 256;
+const kColorAttachmentFormat = 'rgba32float';
+
+// renders texture a slanted plane placed in a specific way
+class DerivativesTest extends GPUTest {
+  copyRenderTargetToBuffer(rt: GPUTexture): GPUBuffer {
+    const byteLength = kRTSize * kBytesPerRow;
+    const buffer = this.device.createBuffer({
+      size: byteLength,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+    });
+
+    const commandEncoder = this.device.createCommandEncoder();
+    commandEncoder.copyTextureToBuffer(
+      { texture: rt, mipLevel: 0, origin: [0, 0, 0] },
+      { buffer, bytesPerRow: kBytesPerRow, rowsPerImage: kRTSize },
+      { width: kRTSize, height: kRTSize, depthOrArrayLayers: 1 }
+    );
+    this.queue.submit([commandEncoder.finish()]);
+
+    return buffer;
+  }
+
+  private pipeline: GPURenderPipeline | undefined;
+  async init(): Promise<void> {
+    await super.init();
+
+    this.pipeline = this.device.createRenderPipeline({
+      vertex: {
+        module: this.device.createShaderModule({
+          code: `
+            struct Outputs {
+              [[builtin(position)]] Position : vec4<f32>;
+              [[location(0)]] fragUV : vec2<f32>;
+            };
+
+            [[stage(vertex)]] fn main(
+              [[builtin(vertex_index)]] VertexIndex : u32) -> Outputs {
+              // Full screen quad
+              var position : array<vec3<f32>, 6> = array<vec3<f32>, 6>(
```

Comment:
You're absolutely right. As I dug into this, I was indeed only getting the right fragment values for the top-left coord of the quad pixels. Am now rendering into an 8x8 render target (this can be increased to anything, really, by setting kRTSize).